lib/omniauth/strategies/bolt.rb

```diff
@@ -39,14 +39,26 @@ def request_phase
 
       def callback_phase
         payload = {
-          grant_type: 'authorization_code',
-          code: session['authorization_code'],
           client_id: options['publishable_key'],
-          # scope: 'openid+bolt.account.manage',
-          scope: session['scope'],
           client_secret: options['api_key']
         }
 
+        token_and_scope = if session['refresh_token']
+                            {
+                              grant_type: 'refresh_token',
+                              refresh_token: session['refresh_token'],
+                              scope: session['refresh_token_scope']
+                            }
+                          else
+                            {
+                              grant_type: 'authorization_code',
+                              code: session['authorization_code'],
+                              scope: session['scope']
+                            }
+                          end
+
+        payload.merge!(token_and_scope)
```

Comment:
I don't understand the purpose of this change.
Here the user is logged in, and a new `session['authorization_code']` is provided [here](https://github.com/nebulab/omniauth-bolt/pull/2/commits/acdcdc3b08cb534d9c94a60125ae6c7aa3c46b99#diff-c0c57b12123a23b6cbe91332a48b0a8eda6ed6ed3c454f6e0e279445cd79f437L34), so the access_token will be never expired here.

Reply:
reworked logic in https://github.com/nebulab/omniauth-bolt/pull/2/commits/9b3154e439f4f511d1c1335517c5b9ba666567ab